File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;

        int i = 2;
        int k = 2;

        while (i < nums.length){
            if(nums[i] != nums[k-2]){
                nums[k] = nums[i];
                k++;
            }
            i++;
        }
        return k;
        
    }
}

File: Data Structures & Algorithms/find-pivot-index/submission-0.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        
        vector<int> prefix_;
        int total = 0;

        for (int n: nums){
            total += n;
            prefix_.push_back(total);
        }

        for (int pivot = 0; pivot < prefix_.size(); pivot++){
            int leftSum = pivot == 0 ? 0 : prefix_[pivot-1];
            int rightSum = prefix_.back() - prefix_[pivot];
            if (leftSum != rightSum){
                continue;
            }
            else {
                return pivot;
            }
        }
        return -1 ;
    }
};